Stopwatch.tsx

```diff
@@ -1,88 +1,112 @@
-                        onClick={this.handleStartClick}>start</button>
-                    : <button type="button" className="stop-btn"
-                        onClick={this.handleStopClick}>stop</button>
-                )}
-                {(secondsElapsed !== 0 && this.incrementer !== lastClearedIncrementer
-                    ? <button type="button" onClick={this.handleLabClick}>lap</button>
-                    : null
-                )}
-                {(secondsElapsed !== 0 && this.incrementer === lastClearedIncrementer
-                    ? <button type="button" onClick={this.handleResetClick}>reset</button>
-                    : null
-                )}
-                <div className="stopwatch-laps">
-                    {this.laps && this.laps.map((lap, i) =>
-                        <Lap index={i + 1} lap={lap} onDelete={this.handleDeleteClick(i)} />)}
-                </div>
+    };
+
+    return (
+        <div className="stopwatch">
+            <h1 className="stopwatch-timer">{formattedSeconds(secondsElapsed)}</h1>
+            {secondsElapsed === 0 ||
+                incrementerRef.current === lastClearedIncrementer ? (
+                <button type="button" className="start-btn" onClick={handleStartClick}>
+                    start
+                </button>
+            ) : (
+                <button type="button" className="stop-btn" onClick={handleStopClick}>
+                    stop
+                </button>
+            )}
+            {secondsElapsed !== 0 &&
+                incrementerRef.current !== lastClearedIncrementer ? (
+                <button type="button" onClick={handleLabClick}>
+                    lap
+                </button>
+            ) : null}
+            {secondsElapsed !== 0 &&
+                incrementerRef.current === lastClearedIncrementer ? (
+                <button type="button" onClick={handleResetClick}>
+                    reset
+                </button>
+            ) : null}
+            <div className="stopwatch-laps">
+                {laps?.length
+                    ? laps.map((lap, i) => (
+                        <Lap
+                            key={`lap-${i}-${lap}`}
+                            index={i}
+                            lap={lap}
+                            onDelete={handleDeleteClick(i)}
+                        />
+                    ))
+                    : null}
             </div>
-        );
-    }
-}
-const Lap = (props: { index: number; lap: number; onDelete: () => {} }) => (
-    <div key={props.index} className="stopwatch-lap">
-        <strong>{props.index}</strong>/ {formattedSeconds(props.lap)}{" "}
-        <button onClick={props.onDelete}> X </button>
+        </div>
+    );
+};
+
+const Lap: React.FC<{
+    index: number;
+    lap: number;
+    onDelete: (index: number) => void;
+}> = ({ index, lap, onDelete }) => (
+    <div key={index} className="stopwatch-lap">
+        <strong>{index}</strong>/ {formattedSeconds(lap)}{" "}
```

Comment:
if we'd prefer to label the laps starting at 1, we would want to do that here:

```suggestion
        <strong>{index + 1}</strong>/ {formattedSeconds(lap)}{" "}
```